include/picongpu/fields/currentDeposition/Esirkepov/BitPacking.hpp

```diff
@@ -0,0 +1,76 @@
+/* Copyright 2013-2022 Axel Huebl, Heiko Burau, Rene Widera, Sergei Bastrakov
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "picongpu/fields/currentDeposition/Esirkepov/Line.hpp"
+
+#include <cstdint>
+
+
+namespace picongpu
+{
+    namespace currentSolver
+    {
+        /** Helper base class for Esirkepov to manage bit packing operations to encode the status of the start and end
+         * particle to optimize the algorithm.
+         * Bit-packing for the status is required to keep the register footprint low.
+         */
+        struct Bitpacking
+        {
+            enum class Status
+            {
+                //! Marker that a particle trajectory is crossing the assignment cell border.
+                leaveCell = 1,
+                //! Marker that the start particle is located in the base assignment cell.
+                startParticleInAssignmentCell = 2,
+                //! Marker that the end particle is located in the base assignment cell.
+                endParticleInAssignmentCell = 4
```

Comment:
I think we normally capitalize those? (I have no preference, just to ensure uniform style)

Reply:
Oh i meant not all caps, but e.g. `LeaveCell`. But since we do not have a fixed style, i'm fine with anything.